x-pack/legacy/plugins/graph/public/helpers/saved_workspace_utils.ts

```diff
@@ -0,0 +1,209 @@
+      perPage: size,
+      searchFields: ['title^3', 'description'],
+    })
+    .then(resp => {
+      return {
+        total: resp.total,
+        hits: resp.savedObjects.map(hit => mapHits(hit, urlFor(basePath, hit.id))),
+      };
+    });
+}
+
+export async function getSW(
+  {
+    savedObjectsClient,
+    indexPatterns,
+  }: { savedObjectsClient: SavedObjectsClientContract; indexPatterns: IndexPatternsContract },
+  id?: string
+) {
+  const savedObject = {
+    id,
+    copyOnSave: false,
+    isSaving: false,
+    // NOTE: this.type (not set in this file, but somewhere else) is the sub type, e.g. 'area' or
+    // 'data table', while esType is the more generic type - e.g. 'visualization' or 'saved search'.
+    getEsType: () => config.type || '',
+    // Overwrite the default getDisplayName function which uses type and which is not very
+    // user friendly for this object.
+    getDisplayName: () => 'graph workspace',
+  };
+
+  if (!id) {
+    assign(savedObject, defaults);
+    // no need for Graph
+    // await hydrateIndexPattern(config.id || '', savedObject as any, indexPatterns, config);
+    // if (typeof config.afterESResp === 'function') {
+    //    savedObject = await config.afterESResp(savedObject);
+    // }
+    return Promise.resolve(savedObject);
+  }
+
+  const resp = await savedObjectsClient.get(savedWorkspaceType, id);
+
+  const respMapped = {
+    _id: resp.id,
+    _type: resp.type,
+    _source: cloneDeep(resp.attributes),
+    references: resp.references,
+    found: !!resp._version,
+  };
+
+  return await applyESRespUtil(indexPatterns, respMapped, savedObject, config);
+
+  // no need for Graph
+  // if (typeof config.init === 'function') {
+  //   await config.init.call(savedObject);
+  // }
+}
+
+export function deleteWS(savedObjectsClient: SavedObjectsClientContract, ids: string[]) {
+  return Promise.all(ids.map((id: string) => savedObjectsClient.delete(savedWorkspaceType, id)));
+}
+
+export async function saveWS(
+  savedObject: SavedObject,
+  {
+    confirmOverwrite = false,
+    isTitleDuplicateConfirmed = false,
+    onTitleDuplicate,
+  }: SavedObjectSaveOpts = {},
+  services: {
+    savedObjectsClient: SavedObjectsClientContract;
+    overlays: OverlayStart;
+  }
+) {
+  // Save the original id in case the save fails.
+  const originalId = savedObject.id;
+  // Read https://github.com/elastic/kibana/issues/9056 and
+  // https://github.com/elastic/kibana/issues/9012 for some background into why this copyOnSave variable
+  // exists.
+  // The goal is to move towards a better rename flow, but since our users have been conditioned
+  // to expect a 'save as' flow during a rename, we are keeping the logic the same until a better
+  // UI/UX can be worked out.
+  if (savedObject.copyOnSave) {
+    delete savedObject.id;
+  }
+
+  // Here we want to extract references and set them within "references" attribute
+  let { attributes, references } = serializeSavedObject(savedObject, config);
```

Comment:
do we need a global `serializeSavedObject`? Seems like this can happen completely within the graph code (just copy everything over and call JSON stringify on the workspace state) as the generic helper doesn't do anything helpful for the Graph case

Reply:
Agree. Fixed.